### PR TITLE
Go 1.7.6

### PR DESCRIPTION
**Pull request by iwankgb**
Fixes issue of using insecure Go version.

Summary of changes:
- Go version updated

Testing done:
- all existing tests should pass.

### CODE REVIEW

**Comment by iwankgb** [comment]
run integration tests

**Comment by iwankgb** [comment]
run integration tests

**Comment by iwankgb** [comment]
run integration tests

File: .travis.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ services:
 language: go
 sudo: required
 go:
-- 1.7
+- 1.7.x
```
Comment (skonefal) [inline .travis.yml]: Is it OK for you to have different version of golang in Travis and in Integration Tests?
We should rather have the same version everywhere, so we should pick just one from:
a) Latest minor wildcard 
b) Specific version

Currently, we have both :(

Comment (iwankgb) [inline .travis.yml]: @skonefal - I assume that we should be running latest version from chosen branch anyway and this approach allows to do so. If we were to add similar automation to vagrant provisioning then we would have to code it or use `gimme` as Travis does.
Side note: 1.7.6 was not available on Travis last night.

**Comment by skonefal** [comment]
run integration tests

**Pull request merged**